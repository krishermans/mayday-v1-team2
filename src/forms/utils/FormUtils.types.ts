import FieldType from "../../types/FieldType";

export interface DisplayableCategory {
  content: string;
  id?: string;
  label: string;
  type: FieldType;
}
