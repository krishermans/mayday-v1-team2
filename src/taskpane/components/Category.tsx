import * as React from 'react';
import { useState } from 'react';
import Section from './Section';
import { mergeStyleSets } from '@fluentui/react/lib/Styling';
import { IconButton } from '@fluentui/react/lib/Button';

// styles for category components
const categoryClassNames = mergeStyleSets({
  categoryHeader: {
    backgroundColor: '#eff6fc', // light blue background
    fontSize: '16px',
    fontWeight: '600',
    color: '#005a9e', // Word's blue color
    display: 'flex',
    alignItems: 'center',
    justifyContent: 'space-between',
    cursor: 'pointer',
    padding: '10px 20px',
    borderBottom: '1px solid #cccccc', // light grey border
  },
  categoryContent: {
    padding: '10px 20px',
  },
  arrowIcon: {
    fontSize: '12px',
    color: '#005a9e', // Word's blue color
  },
});

// category component with expandable/collapsible sections
interface CategoryProps {
  name: string;
  sections: number;
}

const Category: React.FC<CategoryProps> = ({ name, sections }) => {
  const [isOpen, setIsOpen] = useState(false);
  return (
    <div>
      <div
        className={categoryClassNames.categoryHeader}
        onClick={() => setIsOpen(!isOpen)}
      >
        {name} ({sections})
        <IconButton
          iconProps={{ iconName: isOpen ? 'ChevronUp' : 'ChevronDown' }}
          className={categoryClassNames.arrowIcon}
          onClick={() => setIsOpen(!isOpen)}
        />
      </div>
      {isOpen && (
        <div className={categoryClassNames.categoryContent}>
          {[...Array(sections)].map((_, index) => (
            <Section key={index} />
          ))}
        </div>
      )}
    </div>
  );
};

export default Category;
