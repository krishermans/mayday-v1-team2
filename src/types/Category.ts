import SubCategory from "./SubCategory";

interface Category {
  code: Nullable<string>
  colour?: string
  id?: string
  subCategories?: SubCategory[]
  title?: string
}

export default Category;
