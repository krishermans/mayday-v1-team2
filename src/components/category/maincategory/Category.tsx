import * as React from 'react';
import { useState, useRef } from 'react';
import { IconButton } from '@fluentui/react/lib/Button';
import Section from "../subcategory/Section";
import { CategoryProps } from "./Category.types";
import { categoryClassNames } from './Category.styles';
import { ContextMenu } from "../../index";

const Category: React.FC<CategoryProps> = ({ id, name, sections }) => {
  const [isOpen, setIsOpen] = useState(false);
  const colorSquareRef = useRef<HTMLDivElement>(null);

  return (
    <div>
      <ContextMenu trigger={
        <div
          id={`category-${id}`}
          className={categoryClassNames.categoryHeader}
          onClick={() => setIsOpen(!isOpen)}
        >
          <IconButton
            iconProps={{ iconName: isOpen ? 'ChevronDown' : 'ChevronRight' }}
            className={categoryClassNames.arrowIcon}
            onClick={() => setIsOpen(!isOpen)}
          />
          {name} ({sections})
          <div
            ref={colorSquareRef}
            className={categoryClassNames.colorSquare}
            style={{ backgroundColor: '#ff0' }}
          />
        </div>
      }
       menuItems={
          [
            {
              handler: () => {
                console.log(`Edit category ${id}`);
              },
              label: "Bewerken"
            },
            {
              handler: () => {
                console.log(`Delete category ${id}`);
              },
              label: "Verwijderen"
            }
          ]
       }
      />
      {isOpen && (
        <div className={categoryClassNames.categoryContent}>
          {[...Array(sections)].map((_, index) => (
            <Section key={index} />
          ))}
        </div>
      )}
    </div>
  );
};

export default Category;
