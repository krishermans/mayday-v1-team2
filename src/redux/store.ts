import { configureStore } from '@reduxjs/toolkit';
import modalReducer, { initialState as DEFAULT_MODAL_STATE } from "./modal/modal.slice";
import categoryReducer, { initialState as DEFAULT_CATEGORY_STATE } from "./category/category.slice";
import contrastWarningReducer, {
    initialState as DEFAULT_CONTRAST_WARNING_ALERT_STATE
} from "./contrastwarningalert/contrastwarningalert.slice";
import settingsReducer, { initialState as DEFAULT_SETTINGS_STATE } from "./settings/settings.slice";
import { State } from "./store.types";

const store = configureStore({
    middleware: (getDefaultMiddleware) => getDefaultMiddleware(),
    reducer: {
        category: categoryReducer,
        contrastWarningAlert: contrastWarningReducer,
        modal: modalReducer,
        settings: settingsReducer
    }
});

export default store;
export type AppStore = typeof store;
export type RootState = ReturnType<typeof store.getState>;
export type AppDispatch = typeof store.dispatch;

export const initialState: State = {
    category: DEFAULT_CATEGORY_STATE,
    contrastWarningAlert: DEFAULT_CONTRAST_WARNING_ALERT_STATE,
    modal: DEFAULT_MODAL_STATE,
    settings: DEFAULT_SETTINGS_STATE
};