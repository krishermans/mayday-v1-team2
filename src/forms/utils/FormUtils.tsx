import { Input, Label, TableCell, TableRow, Text, Textarea } from "@fluentui/react-components";
import * as React from "react";
import { MutableRefObject } from "react";
import Category from "../../types/Category";
import SubCategory from "../../types/SubCategory";
import { isCategory } from "../../types/IsType";
import FieldType from "../../types/FieldType";
import { ColourPickerComponent } from "../../components";
import { DisplayableCategory } from "./FormUtils.types";

export const formatData = (data: Category | SubCategory, categoryTitle?: string) => {
    if (isCategory(data)) {
        return [
            {
                content: data.title,
                id: "title",
                label: "Titel:",
                type: FieldType.INPUT
            },
            {
                content: data.code,
                id: "code",
                label: "Code:",
                type: FieldType.INPUT
            },
            {
                content: data.colour,
                id: "colour",
                label: "Kleur:",
                type: FieldType.COLOUR_INPUT
            }
        ];
    }
    return [
        {
            content: `[${categoryTitle || "Categorie onbekend"}]`,
            id: "category",
            label: "Categorie:",
            type: FieldType.TEXT
        },
        {
            content: data.description || "",
            id: "description",
            label: "Beschrijving:",
            type: FieldType.TEXTAREA
        }
    ];
};

const renderCell = (type: FieldType, id: string, content: string) => {
    let cell;
    switch (type) {
        case FieldType.LABEL:
            cell = <Label htmlFor={`${id}-input`}>{content}</Label>;
            break;
        case FieldType.TEXT:
            cell = <Text>{content}</Text>;
            break;
        case FieldType.INPUT:
            cell = <Input appearance="underline" required type="text" id={`${id}-input`} defaultValue={content}/>;
            break;
        case FieldType.TEXTAREA:
            cell = <Textarea appearance="outline" required id={`${id}-input`} defaultValue={content}/>;
            break;
        case FieldType.COLOUR_INPUT:
            cell = <ColourPickerComponent/>;
    }

    return <TableCell style={{ width: type === FieldType.LABEL && '20%' }}>{cell}</TableCell>;
};

export const renderRow = (data: DisplayableCategory) =>
    (<TableRow appearance="brand" key={data.id}>
        {renderCell(FieldType.LABEL, data.id, data.label)}
        {renderCell(data.type, data.id, data.content)}
    </TableRow>);

export const getFormField = (
    formRef: MutableRefObject<HTMLFormElement>,
    inputId: string
): RadioNodeList | Element | null => {
    return formRef?.current?.elements.namedItem(`${inputId}-input`);
};

export const readFormField = (
    formRef: MutableRefObject<HTMLFormElement>,
    inputId: string
): string => {
    const inputElement = getFormField(formRef, inputId) as HTMLInputElement;
    return inputElement.value;
};