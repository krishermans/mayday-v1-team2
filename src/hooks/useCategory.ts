import { MutableRefObject } from "react";
import { useAppDispatch, useAppSelector } from "../redux/hooks";
import { selectCategory, selectCreate, selectSubCategory } from "../redux/modal/modal.slice";
import { saveCategory, saveSubCategory, updateCategory, updateSubCategory } from "../middleware/modal/ModalMiddleware";
import { readFormField } from "../forms/utils/FormUtils";
import { isCategory, isSubCategory } from "../types/IsType";
import { selectColour } from "../redux/category/category.slice";
import { isLowContrast } from "../utils/ContrastUtils";
import { getColorObject } from "../utils/ColorUtils";
import { openContrastWarningAlert } from "../middleware/contrastwarningalert/ContrastWarningAlertMiddleware";


const useCategory = () => {

    const dispatch = useAppDispatch();

    const category = useAppSelector(selectCategory);
    const colour = useAppSelector(selectColour);
    const create = useAppSelector(selectCreate);
    const subCategory = useAppSelector(selectSubCategory);

    const data = subCategory ? subCategory : category;

    const handleSubmit = async (formRef: MutableRefObject<HTMLFormElement>) => {
        if (colour) {
            const colorObject = getColorObject(colour);
            if (isLowContrast(colorObject)) {
                dispatch(openContrastWarningAlert());
                return;
            }
        }

        await saveData(formRef);
    };

    const saveData = async (formRef: MutableRefObject<HTMLFormElement>) => {
        if (isCategory(data)) {
            const category = {
                code: readFormField(formRef, "code"),
                colour,
                title: readFormField(formRef, "title")
            }
            if (create) {
                dispatch(saveCategory(category));
            } else {
                dispatch(updateCategory(data.id, category));
            }
        }
        if (isSubCategory(data)) {
            const subCategory = {
                categoryId: data.categoryId,
                description: readFormField(formRef, "description"),
                isFavorite: false
            };
            if (create) {
                dispatch(saveSubCategory(subCategory));
            } else {
                dispatch(updateSubCategory(data.id, subCategory));
            }
        }
    };

    return { categoryTitle: category?.title, data, handleSubmit, saveData };
}

export default useCategory;