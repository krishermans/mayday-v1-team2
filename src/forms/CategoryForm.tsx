import * as React from "react";
import { ReactElement, useEffect, useRef, useState } from "react";
import {
  Button,
  DialogActions,
  DialogBody,
  DialogContent,
  DialogTitle,
  DialogTrigger,
  makeStyles,
  Table,
  TableBody
} from "@fluentui/react-components";
import { useAppDispatch, useAppSelector } from "../redux/hooks";
import { selectTitle } from "../redux/modal/modal.slice";
import { formatData, getFormField, readFormField, renderRow } from "./utils/FormUtils";
import useCategory from "../hooks/useCategory";
import { DisplayableCategory } from "./utils/FormUtils.types";
import { closeModal } from "../middleware/modal/ModalMiddleware";
import {ContrastWarning} from "../components/colourpicker/ContrastWarning";

const useStyles = makeStyles({
  content: {
    display: "flex",
    flexDirection: "column",
    rowGap: "10px"
  }
});

const CategoryForm = (): Nullable<ReactElement> => {

  const [disabled, setDisabled] = useState(true);

  const styles = useStyles();

  const dispatch = useAppDispatch();

  const title = useAppSelector(selectTitle);
  const formRef = useRef<HTMLFormElement>(null);
  const [ showDialog, setShowDialog ] = useState(false);
  const { categoryTitle, data, handleSubmit, saveData } = useCategory(setShowDialog);

  const validate = () => {
    let disabled = false;
    if (data) {
      Object.keys(data).forEach(key => {
        if (getFormField(formRef, key) && !readFormField(formRef, key)) {
          disabled = true;
        }
      });
    }

    setDisabled(disabled);
  };

  useEffect(() => {
    validate();
  }, []);

  // TODO: Close should reset form
  const handleClose = () => dispatch(closeModal());

  if (!data) {
    return null;
  }

  return (
    <form ref={formRef} onChange={validate}>
      <DialogBody>
        <DialogTitle action={null}
                     style={{ textDecoration: "underline" }}>{ title }</DialogTitle>
        <DialogContent className={styles.content}>
          <Table>
            <TableBody>
              {formatData(data, categoryTitle).map((data: DisplayableCategory) => renderRow(data))}
            </TableBody>
          </Table>
        </DialogContent>
        <DialogActions>
          <DialogTrigger disableButtonEnhancement>
            <Button appearance="secondary" onClick={handleClose}>Sluiten</Button>
          </DialogTrigger>
          <Button appearance="primary" onClick={() => handleSubmit(formRef)} disabled={disabled}>
            Bevestigen
          </Button>
        </DialogActions>
      </DialogBody>
      {showDialog && (
        <ContrastWarning saveData={() => saveData(formRef) } showDialog={showDialog} setShowDialog={setShowDialog}/>
          )}
    </form>
  );
};

export default CategoryForm;
