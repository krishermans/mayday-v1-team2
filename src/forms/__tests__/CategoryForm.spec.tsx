import useCategory from "../../hooks/useCategory";
import SubCategory from "../../types/SubCategory";
import { fireEvent } from "@testing-library/react";
import React from "react";
import CategoryForm from "../CategoryForm";
import { renderWithProviders } from "../../__tests__/utils/TestUtils";
import { initialState } from "../../redux/store";
import Category from "../../types/Category";

jest.mock("../../hooks/useCategory");

describe("CategoryForm Test Suite", () => {
  const subCategory: SubCategory = {
    categoryId: "testCategoryId",
    description: "testDescription",
    id: "testId"
  };
  const handleSubmit = jest.fn();
  const useCategoryMock = useCategory as unknown as jest.Mock;
  const setupUseCategoryMock = (data?: SubCategory | Category) => {
    useCategoryMock.mockImplementation(() => ({
      categoryTitle: "Test Title",
      data,
      handleSubmit
    }))
  };

  afterEach(() => {
    handleSubmit.mockReset();
    useCategoryMock.mockReset();
  });

  test("initial render", () => {
    setupUseCategoryMock(subCategory);
    const { container } = renderWithProviders(<CategoryForm />, { preloadedState: initialState });

    expect(container).toMatchSnapshot();
  });

  test("No data", () => {
    setupUseCategoryMock();
    const { container } = renderWithProviders(<CategoryForm />, { preloadedState: initialState });

    expect(container).toMatchSnapshot();
  });

  test("handleSubmit", () => {
    setupUseCategoryMock({ categoryId: "testCategoryId", description: null });
    const { container, getByText } = renderWithProviders(<CategoryForm />, { preloadedState: initialState });

    expect(getByText("Bevestigen")).toBeDisabled();

    fireEvent.change(container.querySelector("textarea#description-input"), { target: { value: "test" } });
    expect(getByText("Bevestigen")).not.toBeDisabled();

    fireEvent.click(getByText("Bevestigen"));
    expect(handleSubmit).toHaveBeenCalledTimes(1);
  });

  test("handleClose", () => {
    const mockDispatch = jest.fn();
    jest.spyOn(require("../../redux/hooks"), "useAppDispatch").mockImplementation(() => mockDispatch);
    jest.spyOn(require("../../middleware/modal/ModalMiddleware"), "closeModal").mockImplementation(() => ({ type: "closeModal" }));
    setupUseCategoryMock(subCategory);
    const { getByText } = renderWithProviders(<CategoryForm />, { preloadedState: initialState });

    fireEvent.click(getByText("Sluiten"));
    expect(mockDispatch).toHaveBeenCalledTimes(1);
    expect(mockDispatch.mock.calls[0][0].type).toEqual( "closeModal");
  });

});
