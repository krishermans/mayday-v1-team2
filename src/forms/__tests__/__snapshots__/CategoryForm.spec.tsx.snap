// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CategoryForm Test Suite No data 1`] = `<div />`;

exports[`CategoryForm Test Suite initial render 1`] = `
<div>
  <form>
    <div
      class="fui-DialogBody r71plkv"
    >
      <h2
        class="fui-DialogTitle rztv7rx ___mh25s00_kwrsjh0 fsyjsko"
        style="text-decoration: underline;"
      />
      <div
        class="fui-DialogContent r1e0mpcm ___5zz7g20_1dlhfox f22iagw f1vx9l62 fd7ph1c"
      >
        <table
          class="fui-Table ___422ui00_e7lxxb0 fgkb47j fhovq9v f1w4nmp0 fmrv4ls fly5x3f f14m3nip"
        >
          <tbody
            class="fui-TableBody ___1gw3ddd_19kkhse f1tp1avn"
          >
            <tr
              class="fui-TableRow ___1o398xn_1gb0u4b f19n0e5 f1ewtqcl f1jazu75 f1xeqee6 fz36nt7 f9znhxp fqrak0z f1o2ludy f1kjnpwc fxmnebo f1witrsb f7tkmfy f1r2dosr f15ngxrw fjbbrdp f1t94bn6 feu1g3u f1uorfem fw60kww f1vxd6vx fg706s2 fnzw4c6 f1u0rzck f16xkysk f1bh3yvw fmi79ni f11fozsx fqlf3fd f9dpb3h f1yat0gj fw2muls f7nae3y fkbere7 fmtyzcc fa97sf3 fb6zhgp fyowp6c f9dii88 fz08sq8 f1afxoft flqq2yx f1b5xrmd f1831rx6 fd94n53 f1e0wld5"
            >
              <td
                class="fui-TableCell ___18c4fhz_1qt9bvz f10pi13n f1nbblvp f1vdfbxk f1ov4xf1 f1f5gg8d fz36nt7 f9znhxp fqrak0z f1o2ludy f1kjnpwc fxmnebo f1witrsb f15pt5es fmrv4ls f1ft4266"
                style="width: 20%;"
              >
                <label
                  class="fui-Label ___1yevxz2_yljzsa0 fk6fouc f19n0e5 fkhj508 f1i3iumi"
                  for="category-input"
                >
                  Categorie:
                </label>
              </td>
              <td
                class="fui-TableCell ___18c4fhz_1qt9bvz f10pi13n f1nbblvp f1vdfbxk f1ov4xf1 f1f5gg8d fz36nt7 f9znhxp fqrak0z f1o2ludy f1kjnpwc fxmnebo f1witrsb f15pt5es fmrv4ls f1ft4266"
              >
                <span
                  class="fui-Text ___nzdt000_bc09y70 fk6fouc fkhj508 f1i3iumi figsok6 fpgzoln f1w7gpdv f6juhto f1mtd64y f1y7q3j9 f2jf649"
                >
                  [Test Title]
                </span>
              </td>
            </tr>
            <tr
              class="fui-TableRow ___1o398xn_1gb0u4b f19n0e5 f1ewtqcl f1jazu75 f1xeqee6 fz36nt7 f9znhxp fqrak0z f1o2ludy f1kjnpwc fxmnebo f1witrsb f7tkmfy f1r2dosr f15ngxrw fjbbrdp f1t94bn6 feu1g3u f1uorfem fw60kww f1vxd6vx fg706s2 fnzw4c6 f1u0rzck f16xkysk f1bh3yvw fmi79ni f11fozsx fqlf3fd f9dpb3h f1yat0gj fw2muls f7nae3y fkbere7 fmtyzcc fa97sf3 fb6zhgp fyowp6c f9dii88 fz08sq8 f1afxoft flqq2yx f1b5xrmd f1831rx6 fd94n53 f1e0wld5"
            >
              <td
                class="fui-TableCell ___18c4fhz_1qt9bvz f10pi13n f1nbblvp f1vdfbxk f1ov4xf1 f1f5gg8d fz36nt7 f9znhxp fqrak0z f1o2ludy f1kjnpwc fxmnebo f1witrsb f15pt5es fmrv4ls f1ft4266"
                style="width: 20%;"
              >
                <label
                  class="fui-Label ___1yevxz2_yljzsa0 fk6fouc f19n0e5 fkhj508 f1i3iumi"
                  for="description-input"
                >
                  Beschrijving:
                </label>
              </td>
              <td
                class="fui-TableCell ___18c4fhz_1qt9bvz f10pi13n f1nbblvp f1vdfbxk f1ov4xf1 f1f5gg8d fz36nt7 f9znhxp fqrak0z f1o2ludy f1kjnpwc fxmnebo f1witrsb f15pt5es fmrv4ls f1ft4266"
              >
                <span
                  class="fui-Textarea ___yvnocc0_ob5ctq0 ftuwxu6 f1ewtqcl f10pi13n f1g0x7ka fhxju0i f1ean75l f1cnd47f f1hu3pq6 f11qmguv f19f4twv f1tyq0we f1aa9q02 f16jpd5f f1jar5jt fyu767a fxugw4r f192inf7 f5tn483 f1vxd6vx f1ojsxk5 fzkkow9 fcdblym fg706s2 fjik90z fj3muxo f1akhkt f1c1zstj f1lxtadh f1gw3sf2 f13zj6fq f1mdlcz9 f1a7op3 f1gboi2j f1cjjd47 ffyw7fx f1kp91vd f1ibwz09 f14pi962 f1lh990p f1jc6hxc fj2g8qd f1yk9hq fhwpy7i f14ee0xe f1xhbsuh fv8e3ye ftb5wc6 fjw5xc1 f1xdyd5c fatpbeo fb7uyps f1cmft4k f1x58t8o f1ibeo51 f1vnc8sk f3e99gv fhljsf7 f7ic3uo f1omjgsz fn6xmsl f1snvl17 f1sn8sm0 f1wovo5e fm0h710 f716mnf fvcxoqz f1ub3y4t f1l4zc64 f1m52nbi fk7lb2a f1knhbbd f15pjvi3 f17itt0b f6ginmj f1grcyuh fk1xjsr fgzu20w fvs00aa f1assf6x f1z0osm6 f4ruux4 fu7v4fk f1fnaxjy f7hodha fwory1w f8jkv7v f1f3jaeo f19usxel fh8au0q f12oevn0 fvdgz8d f1so894s f1cq0lt5"
                >
                  <textarea
                    class="fui-Textarea__textarea ___vxzxgk0_17t7cfl f1ern45e f1n71otn f1h8hb77 f1deefiw f1hu3pq6 f11qmguv f19f4twv f1tyq0we f3rmtva f1ewtqcl f19n0e5 fqerorx fk6fouc f1l02sjl fwyc1cq f13ta7ih f1vw9udw fbb3kq8 f1s6fcnf fvmd9f fp2oml8 f135dnwl f1tdddsa f1e60jzv f1if7ixc fkhj508 figsok6 f1i3iumi f1o1s39h"
                    id="description-input"
                    required=""
                  >
                    testDescription
                  </textarea>
                </span>
              </td>
            </tr>
          </tbody>
        </table>
      </div>
      <div
        class="fui-DialogActions r78gbj ___pj8zjl0_1y84z0u f1a7i8kp fd46tj4 fsyjsko f1f41i0t f1jaqex3 f2ao6jk"
      >
        <button
          class="fui-Button r1alrhcs"
          data-tabster="{"restorer":{"type":1}}"
          type="button"
        >
          Sluiten
        </button>
        <button
          class="fui-Button r1alrhcs ___1akj6hk_ih97uj0 ffp7eso f1p3nwhy f11589ue f1q5o8ev f1pdflbu f1phragk f15wkkf3 f1s2uweq fr80ssc f1ukrpxl fecsdlb f1rq72xc fnp9lpt f1h0usnq fs4ktlq f16h9ulv fx2bmrt f1d6v5y2 f1rirnrt f1uu00uk fkvaka8 f1ux7til f9a0qzu f1lkg8j3 fkc42ay fq7113v ff1wgvm fiob0tu f1j6scgf f1x4h75k f4xjyn1 fbgcvur f1ks1yx8 f1o6qegi fcnxywj fmxjhhp f9ddjv3 f17t0x8g f194v5ow f1qgg65p fk7jm04 fhgccpy f32wu9k fu5nqqq f13prjl2 f1czftr5 f1nl83rv f12k37oa fr96u23"
          type="button"
        >
          Bevestigen
        </button>
      </div>
    </div>
  </form>
</div>
`;